### ISSUE TITLE
Error when trying to get interface for nested inner dataclasses

### ISSUE THREAD

**Issue by richard-moss** [cs-cordero/py-ts-interfaces#9]
Hi there,
I've got a dataclass which is a composition of basic types and dataclasses.  Rather than have to define all the dataclasses at the top level, I'm trying to make them into nested definition.

That's confusing, here is what I mean:

Flat top-level definitions:
```
@dataclass
class Meta(Interface):
    color: str 
    size: float

@dataclass
class Point(Interface):
    x: float
    y: float
    meta: Meta

point1 = Point(x=1, y=2,
               meta=Meta(color='red', size=5.5))
```
Nested definition - where I'm using inner classes:
```
@dataclass
class Point(Interface):

    @dataclass
    class Meta(Interface):
        color: str 
        size: float

    x: float
    y: float
    meta: Meta

point1 = Point(x=1, y=2,
               meta=Point.Meta(color='red', size=5.5))
```
The nested inner class approach works fine, and mypy has no problems with it.

But while the first case will get an interface when using py-ts-interfaces, the second (nested) case wont.  I get an error saying:
```
RuntimeError: Invalid nested Interface reference 'Meta' found for interface Point!
Does 'Meta' exist and is it an Interface?
```

I've tried this both when Meta is inherited from Interface, or not.  In both cases it throws an exception

Any thoughts?

**Comment by cs-cordero** [comment]
Hey thanks for reaching out!  For now the only workaround like you pointed out is to define the dataclasses at the top level.  

This library currently doesn't support inner dataclasses, but I think it should be straightforward to add in.

**Comment by richard-moss** [comment]
Hey - Thanks for the quick reply!

I'm they python guy, not our TS guy - so I've got a quick Q:  I presume TS does support defining such an interface right?

I'll see if we can perhaps add the support in then: I hadn't realised until I tried yesterday that doing inner dataclasses like that was possible.  In some limited testing I did, they seem to work really well - plus it's very easy to sub-class either the top level or embedded classes if you want to use them in an abstract template type model.

I'm quite surprised there is nothing I could find on the web about anyone using them that way!

Anyway - thanks for the great module, and we'll talk on our side about possibly looking to add the support in.

**Comment by richard-moss** [comment]
Hey - Thanks for the quick reply!

I'm they python guy, not our TS guy - so I've got a quick Q:  I presume TS does support defining such an interface right?

I'll see if we can perhaps add the support in then: I hadn't realised until I tried yesterday that doing inner dataclasses like that was possible.  In some limited testing I did, they seem to work really well - plus it's very easy to sub-class either the top level or embedded classes if you want to use them in an abstract template type model.

I'm quite surprised there is nothing I could find on the web about anyone using them that way!

Anyway - thanks for the great module, and we'll talk on our side about possibly looking to add the support in.
